ocaml/xapi/helpers.ml

```diff
@@ -1631,13 +1631,46 @@ let try_internal_async ~__context (marshaller : Rpc.t -> 'b)
           info "try_internal_async: destroying task: t = ( %s )" ref ;
           TaskHelper.destroy ~__context t)
 
+module SystemdService (Name : sig
+  val name : string
+end) : sig
+  val is_active : unit -> bool
+
+  val ensure_enabled_and_restart : unit -> unit
+end = struct
+  let name = Name.name
+
+  let systemctl cmd = call_script !Xapi_globs.systemctl [cmd; name]
+
+  let systemctl_ cmd = systemctl cmd |> ignore
+
+  let is_cmd cmd =
+    let exit_code =
```

Comment:
Is it possible to implement it like this? I believe it would be clearer but I'm not sure about pattern matching in the `exception` case.
```
let exit_code = match systemctl_ cmd with
| () -> 0
| exception Forkhelpers.Spawn_internal_error (_, _, Unix.WEXITED n) -> n
| exception e -> raise e
```